src/Controller/Track.php

```diff
@@ -213,6 +217,74 @@ public function newVersion(Request $request, string $id)
         );
     }
 
+    public function rate(Request $request, $id)
+    {
+        if (!$request->isXmlHttpRequest()) {
+            throw new AccessDeniedHttpException();
+        }
+
+        $em = $this->getDoctrine()
+            ->getManager();
+
+        /**
+         * @var $user \App\Entity\User\User
+         */
+        $user = $this->getUser();
+
+        /**
+         * @var $version Version
+         */
+        $version = $em->getRepository(\App\Entity\Track\Version::class)
+            ->findOneBy(['id' => $id]);
+        if (is_null($version)) {
+            return new JsonResponse(
+                [
+                    'message' => "Invalid track version"
+                ],
+                Response::HTTP_BAD_REQUEST
+            );
+        }
+
+        if ($request->getRealMethod() === 'POST') {
+            $ratingRepo = $em->getRepository(VersionRating::class);
+            /*
+             * Check if user has already submitted rating
+             */
+            $rating = $ratingRepo->findOneBy([
+                'version' => $id,
+                'user' => $user,
+            ]);
+
+            /*
+             * If $rating is null, create new row
+             */
+            if (is_null($rating)) {
+                $rating = new VersionRating();
+                $rating->setUser($user);
+                $rating->setVersion($version);
+            }
+
+            $rating->setRating($request->request->get('rating'));
```

Comment:
`$request->request->get('rating')` should be validated.
At the moment user can alter the html and send ratings like 9, 999, -100